lego/apps/users/models.py

```diff
@@ -495,26 +519,69 @@ class Penalty(BasisModel):
     source_event = models.ForeignKey(
         "events.Event", related_name="penalties", on_delete=models.CASCADE
     )
+    activation_time = models.DateTimeField("date created", null=True, default=None)
 
     objects = UserPenaltyManager()  # type: ignore
 
-    def expires(self):
-        dt = Penalty.penalty_offset(self.created_at) - (
-            timezone.now() - self.created_at
+    def save(self, *args, **kwargs):
+        last_penalty_to_be_expired = Penalty.objects.filter(
+            user=self.user, activation_time__lte=timezone.now()
+        ).order_by("-activation_time")
+
+        self.activation_time = (
+            last_penalty_to_be_expired[0].exact_expiration
+            if len(last_penalty_to_be_expired) != 0
+            else self.created_at
```

Comment:
Do you check if the last penalty to be expired is active? Cause as far as i can see, if you have a penalty that expired half a year ago and then get a new penalty now, the activation time for the new penalty would be set to the expiration of the old one (half a year ago)...

Reply:
Forget it, I see you have filtered by activation_time__lte!